pkg/virt-handler/virtwrap/schema.go

```diff
@@ -85,6 +108,7 @@ type Domain struct {
 
 type DomainStatus struct {
 	Status LifeCycle
+	Reason LifeCycle
```

Comment:
The slight annoyance with splitting the LifeCycle and State(ChangeReason) is that shutdown and shutoff can both be initiated by user. So if we merged the shutdown and poweroff states, and reason was "Unknown", we wouldn't know what the real state is.

Due to this, I believe using untyped constants and 2 different string subtypes is the best choice to express the state & reason.